src/dashboard/script.ts

```diff
@@ -0,0 +1,91 @@
+/// <reference lib="dom" />
+
+import { type FetchPage as IFetchPage } from "./api";
+
+async function FetchPage(page: number): ReturnType<typeof IFetchPage> {
+  const res = await fetch(`/page?page=${page}`);
+  if (!res.ok) return null;
+
+  return res.json();
+}
+
+async function loadUsers(pageCursor: number) {
+  const page = await FetchPage(pageCursor);
+  if (!page) return;
```

Comment:
Loading with test data is incredibly quick, but when we reach production code, loading times might become noticeable. A small loading icon or text would be nice to let the user know the next page is being processed.